addon/-private/system/record-array-manager.js

```diff
@@ -514,6 +358,8 @@ function updateLiveRecordArray(array, internalModels) {
 
   if (modelsToAdd.length > 0)    { array._pushInternalModels(modelsToAdd); }
   if (modelsToRemove.length > 0) { array._removeInternalModels(modelsToRemove); }
+
+  return modelsToAdd || modelsToRemove;
```

Comment:
Why do we return here now?

Reply:
A minor perf optimization for the addon-based filter implementation. This essentially lets us know when none of the internal model changes were an addition or a remove, which tells the addon that it definitely needs to signal a recalculation.  Without this we blindly signal, leading to the occasional double notification.

Reply:
I'll leave a note as to why and also that it can be removed in the follow up refactor to eliminate the unnecessary record-array-manager notifications post 3.5